components/core/src/clp_s/search/OutputHandler.hpp

```diff
@@ -45,14 +47,22 @@ class OutputHandler {
     virtual void write(std::string const& message) = 0;
 
     /**
-     * Flushes the output handler.
+     * Flushes the output handler after each table that gets searched.
      */
     virtual void flush() = 0;
 
+    /**
+     * Performs any final operations after all tables have been searched.
+     */
+    virtual void finish() {}
+
     [[nodiscard]] bool should_output_timestamp() const { return m_should_output_timestamp; }
 
+    [[nodiscard]] bool should_marshal_records() const { return m_should_marshal_records; }
```

Comment:
No one's calling this right now, right? Was the intention to remove it or is there a planned usage for it?